src/docdb/tree/DocDBDocumentTreeItem.ts

```diff
@@ -70,19 +70,23 @@ export class DocDBDocumentTreeItem implements IAzureTreeItem {
     public async update(newData: RetrievedDocument): Promise<RetrievedDocument> {
         const client: DocumentClient = this._collection.getDocumentClient();
         const _self: string = this.document._self;
-        this._document = await new Promise<RetrievedDocument>((resolve, reject) => {
-            client.replaceDocument(_self, newData,
-                { accessCondition: { type: 'IfMatch', condition: newData._etag }, partitionKey: this.partitionKeyValue },
-                (err, updated: RetrievedDocument) => {
-                    if (err) {
-                        reject(new Error(err.body));
-                    } else {
-                        resolve(updated);
-                    }
-                });
-        });
-
-        return this.document;
+        if (Object.getOwnPropertyNames(newData).length === 0) {
+            throw new Error("Updating a document with no fields in the document does nothing");
```

Comment:
I'm not sure this error is very helpful and the user would run into the same problem if they just deleted the _self/_etag fields and left all other fields. Maybe just check if _self and _etag are defined and change the message to something like "The '_self' and '_etag' fields are required to update a document."